src/Features/Core/Portable/AddStaticModifier/AbstractAddStaticModifierCodeFixProvider.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.AddStaticModifier
+{
+    internal abstract class AbstractAddStaticModifierCodeFixProvider
+        : SyntaxEditorBasedCodeFixProvider
+    {
+        protected abstract SyntaxNode MapToDeclarator(SyntaxNode declaration);
+
+        internal sealed override CodeFixCategory CodeFixCategory => CodeFixCategory.Compile;
+
+        public sealed override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            context.RegisterCodeFix(
+                new MyCodeAction(c => FixAsync(context.Document, context.Diagnostics[0], c)),
```

Comment:
Shall I play defensive here and check that there is one diagnostic before accessing it?

Reply:
You should just pass in `context.Diagnostics` here.

Reply:
`FixAsync` has only one overload that takes a single `Diagnostic`.

Reply:
I always use a `foreach (var diagnostic in context.Diagnostics)` for this method, but I think @mavasani implements them with an assumption that only one diagnostic is provided.

Reply:
No, I meant to convey that one should never assume a single diagnostic is reported. Ideally, you should never need to poke at individual diagnostics or register code fix separately for each diagnostic. The code fix context has the relevant text span (which is same for each diagnostic), so finding the relevant syntax node for diagnostic(s) can be done using `CodeFixContext.Span`. Only time one would need to poke at each individual diagnostic would be if the analyzer added specific `Diagnostic.Properties` on the diagnostic instance to aid the code fix provider.

Reply:
@Evangelink I looked at other IDE code fix providers based on SyntaxEditorBasedCodeFixProvider, and they are implemented in the same way as you have done it here, so you should be good.